fili-core/src/test/groovy/com/yahoo/bard/webservice/web/ResponseWriterSpec.groovy

```diff
@@ -0,0 +1,327 @@
+// Copyright 2017 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web
+
+import static com.yahoo.bard.webservice.data.time.DefaultTimeGrain.DAY
+
+import com.yahoo.bard.webservice.application.ObjectMappersSuite
+import com.yahoo.bard.webservice.config.SystemConfig
+import com.yahoo.bard.webservice.config.SystemConfigProvider
+import com.yahoo.bard.webservice.data.Result
+import com.yahoo.bard.webservice.data.ResultSet
+import com.yahoo.bard.webservice.data.ResultSetSchema
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.Dimension
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn
+import com.yahoo.bard.webservice.data.dimension.DimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionRow
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager
+import com.yahoo.bard.webservice.data.metric.LogicalMetric
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.util.JsonSlurper
+import com.yahoo.bard.webservice.util.JsonSortStrategy
+import com.yahoo.bard.webservice.util.Pagination
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.DateTime
+import org.joda.time.DateTimeZone
+
+import spock.lang.Specification
+
+import javax.ws.rs.core.UriBuilder
+
+/**
+ * This class host common fields and functions shared by all ResponseWriterSpec.
+ */
+abstract class ResponseWriterSpec extends Specification {
```

Comment:
At some point, we'll probably also want to add a suite of tests to this class. Then, implementors can write a spec file that extends this one, and Spock will run these tests for that implementation without the customer having to rewrite them.

I'm not going to block on that though. This is adequate for now, and I'm not even sure what kinds of tests we could write, since the whole point is to customize the output!